### ISSUE TITLE
Pretrained model for RSICD

### ISSUE THREAD

**Issue by RicRicci22** [Chen-Yang-Liu/MLAT#2]
Dear Chen-Yang-Liu, 
I'm planning to use the results of your method for a paper I'm writing. 
Can you please provide the pretrained model on RSICD, so that I can run it on a custom image and obtain a caption? 
Thank you for your work,

Riccardo

**Comment by Chen-Yang-Liu** [comment]
> Dear Chen-Yang-Liu, I'm planning to use the results of your method for a paper I'm writing. Can you please provide the pretrained model on RSICD, so that I can run it on a custom image and obtain a caption? Thank you for your work,
> 
> Riccardo

Sorry, due to a long time, the relevant best-trained models and logs are not retained now. According to my impression, the best model will be obtained within 10 epochs. You can try retraining the model in a short time.

**Comment by RicRicci22** [comment]
Thank you for the quick reply! 
I was wondering to do it, but I don't know the actual format that the dataset should have in order to train the model! 
Can you point me to the right format that you used in your work? 
Thank you again, 

Riccardo

**Comment by Chen-Yang-Liu** [comment]
> Thank you for the quick reply! I was wondering to do it, but I don't know the actual format that the dataset should have in order to train the model! Can you point me to the right format that you used in your work? Thank you again,
> 
> Riccardo

You can download the RSICD dataset, which contains `images\xxx.jpg` and `dataset_rsicd.json`. Then you can follow the new readme.md in this repository. I have updated it just now to facilitate your training.

**Comment by RicRicci22** [comment]
Thank you very much! 
I managed to run the training just now :) 
I had to do a little bit of more tricks to get the workers>0 to run on Windows with hp5y, but now it is working! 
I also added persistent_workers=True when the dataloader is created to keep the workers on while switching epoch and speed up training! 
I will update the results I get soon! 

Riccardo

**Comment by RicRicci22** [comment]
hey there, I'm here with some results! I trained the method for 10 epochs and I compute the results on RSICD, this is what I got
BLEU-1 0.6464670375669815 
BLEU-2 0.47963562524622433 
BLEU-3 0.37688866859397135 
BLEU-4 0.30727506167219965 
METEOR 0.2610822983974145 
ROUGE_L 0.4825852448538177 
CIDEr 0.8887841427788188

I see that the results are different compared to the paper, have you any idea why this is possible? 
Thank you and best regards, 

Riccardo

**Comment by Chen-Yang-Liu** [comment]
> hey there, I'm here with some results! I trained the method for 10 epochs and I compute the results on RSICD, this is what I got BLEU-1 0.6464670375669815 BLEU-2 0.47963562524622433 BLEU-3 0.37688866859397135 BLEU-4 0.30727506167219965 METEOR 0.2610822983974145 ROUGE_L 0.4825852448538177 CIDEr 0.8887841427788188
> 
> I see that the results are different compared to the paper, have you any idea why this is possible? Thank you and best regards,
> 
> Riccardo

I am puzzled by the low scores mentioned above. It looks like that's even lower than the baseline model reported in our paper. I have some possible guesses:
1. Please make sure you use the beam search strategy when running eval.py.
2. You can try to calculate the score on the test set for the model saved for each epoch. Because, I found that there is a large data distribution gap between the validation set and the test set.

**Comment by RicRicci22** [comment]
I realized that I was using a beam of 1, instead now I will try with a beam of 5 and also computing it for each model. I will try your suggestions and reply with the results! 

Thank you a lot,

Riccardo

**Comment by zhouting1994** [comment]
> I realized that I was using a beam of 1, instead now I will try with a beam of 5 and also computing it for each model. I will try your suggestions and reply with the results!
> 
> Thank you a lot,
> 
> Riccardo

Hi, did you get the latest results? I reproduced the code and got similar results to yours. I used the best checkpoint on eval.py with a beam of 5 but only got bleu-4 = 0.29, so if you have any improvements, please let me know how to improve.  Thank you so much.

**Comment by hanhan68** [comment]
> > I realized that I was using a beam of 1, instead now I will try with a beam of 5 and also computing it for each model. I will try your suggestions and reply with the results!
> > Thank you a lot,
> > Riccardo
> 
> Hi, did you get the latest results? I reproduced the code and got similar results to yours. I used the best checkpoint on eval.py with a beam of 5 but only got bleu-4 = 0.29, so if you have any improvements, please let me know how to improve. Thank you so much.

请问这个LSTM方法有效吗？我跑出来的效果也很差
